modules/apps/layout/layout-impl/src/main/java/com/liferay/layout/internal/search/spi/model/index/contributor/LayoutModelDocumentContributor.java

```diff
@@ -95,6 +96,12 @@ public void contribute(Document document, Layout layout) {
 
 		if (serviceContext != null) {
```

Comment:
@pavel-savinov I don't have problem with your request wrapping logic. However this serviceContext null check is a very serious problem. In case it is null, the request and response are all left as null. And in the following usages, most of them are not checking for null request/response. We could get NPE.

Reply:
Hi @shuyangzhou, sent [here](https://github.com/brianchandotcom/liferay-portal/pull/69435) with the changes you mentioned. Thanks!